synapse/storage/push_rule.py

```diff
@@ -185,6 +185,63 @@ def bulk_get_push_rules(self, user_ids):
 
         defer.returnValue(results)
 
+    @defer.inlineCallbacks
+    def copy_push_rule_from_room_to_room(
+        self, old_room_id, new_room_id, user_id, rule,
+    ):
+        """Copy a single push rule from one room to another for a specific user.
+
+        Args:
+            old_room_id (str): ID of the old room.
+            new_room_id (str): ID of the new room.
+            user_id (str): ID of user the push rule belongs to.
+            rule (Dict): A push rule.
+        """
+        # Create new rule id
+        rule_id_scope = '/'.join(rule["rule_id"].split('/')[:-1])
+        new_rule_id = rule_id_scope + "/" + new_room_id
+
+        # Change room id in each condition
+        for index, condition in enumerate(rule["conditions"]):
```

Comment:
why not just:

```
for condition in rule["conditions"]:
    if condition["key"] == "room_id":
        condition["pattern"] = new_room_id
```

also: I note that you are checking for the existence of `"key" in condition` in the calling function. Shouldn't we do that here?